src/PowerShellEditorServices/Services/PowerShellContext/Handlers/GetCommentHelpHandler.cs

```diff
@@ -80,21 +79,27 @@ public async Task<CommentHelpRequestResult> Handle(CommentHelpRequestParams requ
                     vscodeSnippetCorrection: true,
                     placement: helpLocation));
 
-            string helpText = analysisResults?.FirstOrDefault()?.Correction?.Edits[0].Text;
+            string helpText = analysisResults?[0]?.Correction?.Edits[0].Text;
 
             if (helpText == null)
             {
                 return result;
             }
 
-            result.Content = ScriptFile.GetLinesInternal(helpText).ToArray();
+            List<string> helpLines = ScriptFile.GetLinesInternal(helpText);
 
             if (helpLocation != null &&
                 !helpLocation.Equals("before", StringComparison.OrdinalIgnoreCase))
             {
                 // we need to trim the leading `{` and newline when helpLocation=="begin"
                 // we also need to trim the leading newline when helpLocation=="end"
-                result.Content = result.Content.Skip(1).ToArray();
+                helpLines = helpLines.GetRange(1, helpLines.Count - 1);
+            }
+
+            // Trim trailing newline from help text.
+            if (string.IsNullOrEmpty(helpLines[helpLines.Count - 1]))
+            {
+                result.Content = helpLines.GetRange(0, helpLines.Count - 1).ToArray();
```

Comment:
Should this be where `result.Content` is assigned?

If there's no empty line at the end, it seems we don't assign `result.Content`.

So maybe instead the way to go is:

```csharp
if (string.IsNullOrEmpty(helpLines[helpLines.Count - 1]))
{
    helpLines.RemoveAt(helpLines.Count - 1);
}

result.Content = helpLines.ToArray();
return result;
```